### PR TITLE
xxhash cleanup and fixes

### PR DESCRIPTION
**Pull request by tristan957**
Trying to diagnose the following compilation issue with `-Doptimization=g`, I found these issues and cleanups.

```
[348/1057] Compiling C object lib/libhse-3.so.3.0.0.p/cn_kblock_builder.c.o
FAILED: lib/libhse-3.so.3.0.0.p/cn_kblock_builder.c.o
cc -Ilib/libhse-3.so.3.0.0.p -Ilib -I../lib -Ilib/include -I../lib/include -Iinclude -I../include -Ilib/util/include -I../lib/util/include -Ilib/mpool/include -I../lib/mpool/include -Isubprojects/xxHash-0.8.1 -I../subprojects/xxHash-0.8.1 -Isubprojects/lz4-1.9.3/lib -I../subprojects/lz4-1.9.3/lib -Isubprojects/cJSON-1.7.15/include-workaround-meson -I../subprojects/cJSON-1.7.15/include-workaround-meson -Isubprojects/hyperloglog -I../subprojects/hyperloglog -Ilib/error/include -I../lib/error/include -Ilib/logging/include -I../lib/logging/include -Ilib/pidfile/include -I../lib/pidfile/include -Isubprojects/rbtree -I../subprojects/rbtree -Isubprojects/crc32c -I../subprojects/crc32c -Isubprojects/xoroshiro -I../subprojects/xoroshiro -I/usr/include/p11-kit-1 -fvisibility=hidden -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=gnu11 -O2 -g -fmacro-prefix-map=../= -DHSE_BUILD_DEBUG -D_GNU_SOURCE -D_LGPL_SOURCE -Wlogical-op -Wno-missing-field-initializers -Wmaybe-uninitialized -Wno-sign-compare -Wno-unused-parameter -DLOG_DEFAULT=7 -DLEVEL1_DCACHE_LINESIZE=64 -DURCU_INLINE_SMALL_FUNCTIONS -Og -g3 -fPIC -pthread -DXXH_INLINE_ALL -DXXH_INLINE_HINTS -DXXH_STATIC_LINKING_ONLY -MD -MQ lib/libhse-3.so.3.0.0.p/cn_kblock_builder.c.o -MF lib/libhse-3.so.3.0.0.p/cn_kblock_builder.c.o.d -o lib/libhse-3.so.3.0.0.p/cn_kblock_builder.c.o -c ../lib/cn/kblock_builder.c
In file included from ../lib/util/include/hse_util/hash.h:14,
                 from ../lib/include/hse_ikvdb/key_hash.h:9,
                 from ../lib/cn/kblock_builder.c:12:
In function ‘XXH3_accumulate’,
    inlined from ‘XXH3_hashLong_internal_loop’ at ../subprojects/xxHash-0.8.1/xxhash.h:4391:9,
    inlined from ‘XXH3_hashLong_64b_internal’ at ../subprojects/xxHash-0.8.1/xxhash.h:4454:5,
    inlined from ‘XXH3_hashLong_64b_withSecret’ at ../subprojects/xxHash-0.8.1/xxhash.h:4474:12,
    inlined from ‘XXH3_64bits_internal’ at ../subprojects/xxHash-0.8.1/xxhash.h:4555:12,
    inlined from ‘XXH_INLINE_XXH3_64bits_withSecret’ at ../subprojects/xxHash-0.8.1/xxhash.h:4571:12:
../subprojects/xxHash-0.8.1/xxhash.h:3932:1: error: inlining failed in call to ‘always_inline’ ‘XXH3_accumulate_512_sse
’: function not considered for inlining
 3932 | XXH3_accumulate_512_sse2( void* XXH_RESTRICT acc,
      | ^~~~~~~~~~~~~~~~~~~~~~~~
../subprojects/xxHash-0.8.1/xxhash.h:4369:9: note: called from here
 4369 |         f_acc512(acc,
      |         ^~~~~~~~~~~~~
 4370 |                  in,
      |                  ~~~
 4371 |                  secret + n*XXH_SECRET_CONSUME_RATE);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘XXH3_hashLong_internal_loop’,
    inlined from ‘XXH3_hashLong_64b_internal’ at ../subprojects/xxHash-0.8.1/xxhash.h:4454:5,
    inlined from ‘XXH3_hashLong_64b_withSecret’ at ../subprojects/xxHash-0.8.1/xxhash.h:4474:12,
    inlined from ‘XXH3_64bits_internal’ at ../subprojects/xxHash-0.8.1/xxhash.h:4555:12,
    inlined from ‘XXH_INLINE_XXH3_64bits_withSecret’ at ../subprojects/xxHash-0.8.1/xxhash.h:4571:12:
../subprojects/xxHash-0.8.1/xxhash.h:3967:1: error: inlining failed in call to ‘always_inline’ ‘XXH3_scrambleAcc_sse2’: function not considered for inlining
 3967 | XXH3_scrambleAcc_sse2(void* XXH_RESTRICT acc, const void* XXH_RESTRICT secret)
      | ^~~~~~~~~~~~~~~~~~~~~
../subprojects/xxHash-0.8.1/xxhash.h:4392:9: note: called from here
 4392 |         f_scramble(acc, secret + secretSize - XXH_STRIPE_LEN);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘XXH3_accumulate’,
    inlined from ‘XXH3_hashLong_internal_loop’ at ../subprojects/xxHash-0.8.1/xxhash.h:4399:9,
    inlined from ‘XXH3_hashLong_64b_internal’ at ../subprojects/xxHash-0.8.1/xxhash.h:4454:5,
    inlined from ‘XXH3_hashLong_64b_withSecret’ at ../subprojects/xxHash-0.8.1/xxhash.h:4474:12,
    inlined from ‘XXH3_64bits_internal’ at ../subprojects/xxHash-0.8.1/xxhash.h:4555:12,
    inlined from ‘XXH_INLINE_XXH3_64bits_withSecret’ at ../subprojects/xxHash-0.8.1/xxhash.h:4571:12:
../subprojects/xxHash-0.8.1/xxhash.h:3932:1: error: inlining failed in call to ‘always_inline’ ‘XXH3_accumulate_512_sse
’: function not considered for inlining
 3932 | XXH3_accumulate_512_sse2( void* XXH_RESTRICT acc,
      | ^~~~~~~~~~~~~~~~~~~~~~~~
../subprojects/xxHash-0.8.1/xxhash.h:4369:9: note: called from here
 4369 |         f_acc512(acc,
      |         ^~~~~~~~~~~~~
 4370 |                  in,
      |                  ~~~
 4371 |                  secret + n*XXH_SECRET_CONSUME_RATE);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘XXH3_hashLong_internal_loop’,
    inlined from ‘XXH3_hashLong_64b_internal’ at ../subprojects/xxHash-0.8.1/xxhash.h:4454:5,
    inlined from ‘XXH3_hashLong_64b_withSecret’ at ../subprojects/xxHash-0.8.1/xxhash.h:4474:12,
    inlined from ‘XXH3_64bits_internal’ at ../subprojects/xxHash-0.8.1/xxhash.h:4555:12,
    inlined from ‘XXH_INLINE_XXH3_64bits_withSecret’ at ../subprojects/xxHash-0.8.1/xxhash.h:4571:12:
../subprojects/xxHash-0.8.1/xxhash.h:3932:1: error: inlining failed in call to ‘always_inline’ ‘XXH3_accumulate_512_sse
’: function not considered for inlining
 3932 | XXH3_accumulate_512_sse2( void* XXH_RESTRICT acc,
      | ^~~~~~~~~~~~~~~~~~~~~~~~
../subprojects/xxHash-0.8.1/xxhash.h:4404:13: note: called from here
 4404 |             f_acc512(acc, p, secret + secretSize - XXH_STRIPE_LEN - XXH_SECRET_LASTACC_START);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[413/1057] Compiling C object cli/hse3.p/cli_util.c.o
ninja: build stopped: subcommand failed.
```

### CODE REVIEW

**Comment by tristan957** [comment]
I guess bumping the version has to be the last commit?

**Comment by eli-schwartz** [comment]
Regardless, the tag will apply to the HEAD commit when merging.

It's not... semantically wrong... to bump a project version, then finalize the release in a followup commit. The wrapdb isn't especially different from that generic project concept, I guess.

File: subprojects/packagefiles/xxhash/meson.build [hunk 1]
```diff
@@ -55,7 +55,10 @@ if get_option('prefetch-dist') > 0
     c_args += '-DXXH_PREFETCH_DIST=@0@'.format(get_option('prefetch-dist'))
 endif
 
-if not get_option('inline-hints').auto()
+if cc.get_id() == 'gcc' and cc.version().version_compare('>=12.0.0') and get_option('optimization') == 'g'
+    c_args += '-DXXH_NO_INLINE_HINTS=1'
+    message('Ignoring value of inline-hints, see https://github.com/Cyan4973/xxHash/pull/720')
+elif not get_option('inline-hints').auto()
```
Comment (beckerg) [inline subprojects/packagefiles/xxhash/meson.build:62]: Does this work if I pass in options with CFLAGS or C_ARGS?

Comment (beckerg) [reply subprojects/packagefiles/xxhash/meson.build:62]: Hmmm, no inline hints, how will that affect the build quality?

Comment (tristan957) [reply subprojects/packagefiles/xxhash/meson.build:66]: Greg, look here: https://github.com/Cyan4973/xxHash/blob/dev/xxhash.h#L1677. That is the behavior if left unspecified.

Comment (tristan957) [reply subprojects/packagefiles/xxhash/meson.build:66]: CFLAGS should get passed down to the build of xxhash. @eli-schwartz can correct me if I am wrong.

Comment (tristan957) [reply subprojects/packagefiles/xxhash/meson.build:66]: We don't set this particular option in HSE, but we do set `inline-all`

Comment (eli-schwartz) [reply subprojects/packagefiles/xxhash/meson.build:66]: If you pass -Og in via CFLAGS then Meson doesn't sync that with `get_option('optimization')` and this meson.build will simply progress on to check the inline-hints option.

So in the case you used CFLAGS here, you'd want to make sure you set inline-hints yourself, or leave it at the default autodetection which may do anything, but won't take compiler versions into account.

**Comment by eli-schwartz** [comment]
Thanks!

**Comment by tristan957** [comment]
Thanks for your help

**Pull request merged**